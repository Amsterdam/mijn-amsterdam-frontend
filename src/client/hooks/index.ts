export * from './analytics.hook';
export * from './detectResize.hook';
export * from './media.hook';
export * from './modalRoot.hook';
export * from './optin.hook';
export * from './pageChange';
export * from './storage.hook';
export * from './timer.hook';
export * from './useScript';
export * from './useSSE';
export * from './useThrottledFn.hook';
