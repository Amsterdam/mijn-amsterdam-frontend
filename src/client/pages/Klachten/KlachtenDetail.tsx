import { useParams } from 'react-router';

import { useKlachtenThemaData } from './useKlachtenThemaData.hook';
import type { Klacht } from '../../../server/services/klachten/types';
import { Datalist, Row, RowSet } from '../../components/Datalist/Datalist';
import { PageContentCell } from '../../components/Page/Page';
import ThemaDetailPagina from '../ThemaPagina/ThemaDetailPagina';

type KlachtenDetailContentProps = {
  klacht: Klacht;
};

function KlachtenDetailContent({ klacht }: KlachtenDetailContentProps) {
  const rows: Array<Row | RowSet> = [
    {
      label: 'Nummer van uw klacht',
      content: klacht?.id || '-',
    },
    {
      label: 'Ontvangen op',
      content: klacht?.ontvangstDatumFormatted,
    },
    {
      label: 'Wat is de klacht?',
      content: klacht?.omschrijving,
    },
    {
      label: 'Wat is de locatie waar de klacht is ontstaan?',
      content: klacht?.locatie,
    },
    {
      label:
        'Wat wilt u dat de gemeente gaat doen naar aanleiding van uw klacht?',
      content: klacht?.gewensteOplossing,
    },
  ].filter((row) => !!row.content);

  return (
    <PageContentCell>
      <Datalist rows={rows} />
    </PageContentCell>
  );
}

export function KlachtenDetailPagina() {
  const { klachten, isLoading, isError, breadcrumbs } = useKlachtenThemaData();
  const { id } = useParams<{ id: string }>();
  const klacht = klachten.find((klacht) => klacht.id === id);

  return (
    <ThemaDetailPagina
      title={klacht?.onderwerp || 'Klacht'}
      zaak={klacht}
      isError={isError}
      isLoading={isLoading}
      pageContentMain={klacht && <KlachtenDetailContent klacht={klacht} />}
      breadcrumbs={breadcrumbs}
    />
  );
}
