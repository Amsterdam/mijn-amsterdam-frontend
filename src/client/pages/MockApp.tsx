import { RecoilRoot, RecoilRootProps } from 'recoil';
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';

interface MockAppProps {
  routePath: string;
  routeEntry: string;
  initializeState?: RecoilRootProps['initializeState'];
  component: React.Component | any;
}

export default function MockApp({
  routePath,
  routeEntry,
  initializeState,
  component,
}: MockAppProps) {
  return (
    <RecoilRoot initializeState={initializeState}>
      <MemoryRouter initialEntries={[routeEntry]}>
        <Route path={routePath} component={component} />
      </MemoryRouter>
    </RecoilRoot>
  );
}
