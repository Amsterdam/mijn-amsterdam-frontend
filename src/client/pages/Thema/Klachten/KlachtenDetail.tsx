import { useParams } from 'react-router';

import { useKlachtenThemaData } from './useKlachtenThemaData.hook.ts';
import type { KlachtFrontend } from '../../../../server/services/klachten/types.ts';
import { Datalist, Row, RowSet } from '../../../components/Datalist/Datalist.tsx';
import { PageContentCell } from '../../../components/Page/Page.tsx';
import ThemaDetailPagina from '../../../components/Thema/ThemaDetailPagina.tsx';
import { useHTMLDocumentTitle } from '../../../hooks/useHTMLDocumentTitle.ts';

type KlachtenDetailContentProps = {
  klacht: KlachtFrontend;
};

function KlachtenDetailContent({ klacht }: KlachtenDetailContentProps) {
  const rows: Array<Row | RowSet> = [
    {
      label: 'Nummer van uw klacht',
      content: klacht?.id || '-',
    },
    {
      label: 'Ontvangen op',
      content: klacht?.ontvangstDatumFormatted,
    },
    {
      label: 'Wat is de klacht?',
      content: klacht?.omschrijving,
    },
    {
      label: 'Wat is de locatie waar de klacht is ontstaan?',
      content: klacht?.locatie,
    },
    {
      label:
        'Wat wilt u dat de gemeente gaat doen naar aanleiding van uw klacht?',
      content: klacht?.gewensteOplossing,
    },
  ].filter((row) => !!row.content);

  return (
    <PageContentCell>
      <Datalist rows={rows} />
    </PageContentCell>
  );
}

export function KlachtenDetail() {
  const { klachten, isLoading, isError, breadcrumbs, routeConfig } =
    useKlachtenThemaData();
  useHTMLDocumentTitle(routeConfig.detailPage);
  const { id } = useParams<{ id: string }>();
  const klacht = klachten.find((klacht) => klacht.id === id);

  return (
    <ThemaDetailPagina
      title={klacht?.onderwerp || 'Klacht'}
      zaak={klacht}
      isError={isError}
      isLoading={isLoading}
      pageContentMain={klacht && <KlachtenDetailContent klacht={klacht} />}
      breadcrumbs={breadcrumbs}
    />
  );
}
