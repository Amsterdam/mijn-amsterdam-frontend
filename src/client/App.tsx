import * as Sentry from '@sentry/browser';
import classnames from 'classnames';
import React, { useContext, useEffect } from 'react';
import ErrorBoundary from 'react-error-boundary';
import {
  BrowserRouter,
  Redirect,
  Route,
  Switch,
  useLocation,
} from 'react-router-dom';
import { AppRoutes, FeatureToggle } from '../universal/config';
import { getOtapEnvItem, IS_PRODUCTION } from '../universal/config/env';
import { isPrivateRoute } from '../universal/helpers';
import styles from './App.module.scss';
import {
  ApplicationError,
  AutoLogoutDialog,
  MainFooter,
  MainHeader,
} from './components';
import { DefaultAutologoutDialogSettings } from './components/AutoLogoutDialog/AutoLogoutDialog';
import {
  TMA_LOGIN_URL_DIGID_AFTER_REDIRECT,
  TMA_LOGIN_URL_EHERKENNING_AFTER_REDIRECT,
} from './config/api';
import {
  useAnalytics,
  useLocalStorage,
  usePageChange,
  useScript,
} from './hooks';
import {
  Burgerzaken,
  BurgerzakenDetail,
  Dashboard,
  GarbageInformation,
  Inkomen,
  InkomenDetail,
  InkomenDetailTozo,
  InkomenSpecificaties,
  LandingPage,
  MyArea,
  MyNotifications,
  MyTips,
  NotFound,
  Profile,
  Zorg,
  ZorgDetail,
  GeneralInfo,
  Vergunningen,
  VergunningDetail,
  Accessibility,
} from './pages';
import { SessionContext, SessionState } from './SessionState';
import { RecoilRoot } from 'recoil';
import ProfileCommercial from './pages/Profile/ProfileCommercial';
import { useAppState } from './hooks/useAppState';

function AppNotAuthenticated() {
  const location = useLocation();

  const [routeEntry, setRouteEntry] = useLocalStorage('RouteEntry', '');

  if (
    (!routeEntry || routeEntry === '/') &&
    location.pathname !== '/' &&
    isPrivateRoute(location.pathname)
  ) {
    setRouteEntry(location.pathname);
  }

  return (
    <>
      <div className={classnames(styles.App, styles.NotYetAuthenticated)}>
        <MainHeader />
        <Switch>
          <Route exact path={AppRoutes.ROOT} component={LandingPage} />
          <Route path={AppRoutes.ACCESSIBILITY} component={Accessibility} />
          <Route
            render={({ location: { pathname } }) => {
              if (isPrivateRoute(pathname)) {
                // Private routes are redirected to Home
                return <Redirect to={AppRoutes.ROOT} />;
              }
              // All other routes are presented with a 404 page
              return <Route component={NotFound} />;
            }}
          />
        </Switch>
      </div>
      <MainFooter />
    </>
  );
}

function AppAuthenticated() {
  useAppState();
  const location = useLocation();
  const session = useContext(SessionContext);
  const [routeEntry, setRouteEntry] = useLocalStorage('RouteEntry', '');

  const redirectAfterLogin = routeEntry || AppRoutes.ROOT;

  usePageChange();

  useEffect(() => {
    if (routeEntry) {
      setRouteEntry('');
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return location.pathname === AppRoutes.BUURT ? (
    <MyArea />
  ) : (
    <>
      <MainHeader isAuthenticated={session.isAuthenticated} />
      <div className={styles.App} id="AppContent">
        <Switch>
          <Redirect
            from={TMA_LOGIN_URL_DIGID_AFTER_REDIRECT}
            to={redirectAfterLogin}
          />
          <Redirect
            from={TMA_LOGIN_URL_EHERKENNING_AFTER_REDIRECT}
            to={redirectAfterLogin}
          />
          <Route exact path={AppRoutes.ROOT} component={Dashboard} />
          <Route path={AppRoutes.NOTIFICATIONS} component={MyNotifications} />
          <Route path={AppRoutes.BRP} component={Profile} />
          <Route path={AppRoutes.KVK} component={ProfileCommercial} />
          <Route path={AppRoutes.TIPS} component={MyTips} />
          <Route
            path={AppRoutes['INKOMEN/STADSPAS']}
            component={InkomenDetail}
          />
          <Route
            path={AppRoutes['INKOMEN/BIJSTANDSUITKERING']}
            component={InkomenDetail}
          />
          <Route
            path={AppRoutes['INKOMEN/SPECIFICATIES']}
            component={InkomenSpecificaties}
          />
          {FeatureToggle.tozoActive && (
            <Route
              path={AppRoutes['INKOMEN/TOZO']}
              component={InkomenDetailTozo}
            />
          )}
          <Route path={AppRoutes.INKOMEN} component={Inkomen} />
          <Route
            path={AppRoutes['ZORG/VOORZIENINGEN']}
            component={ZorgDetail}
          />
          <Route path={AppRoutes.ZORG} component={Zorg} />
          <Route
            path={AppRoutes.BURGERZAKEN_DOCUMENT}
            component={BurgerzakenDetail}
          />
          <Route path={AppRoutes.BURGERZAKEN} component={Burgerzaken} />
          {FeatureToggle.garbageInformationPage && (
            <Route path={AppRoutes.AFVAL} component={GarbageInformation} />
          )}
          <Route path={AppRoutes.ACCESSIBILITY} component={Accessibility} />
          <Route path={AppRoutes.GENERAL_INFO} component={GeneralInfo} />
          <Route
            path={AppRoutes['VERGUNNINGEN/DETAIL']}
            component={VergunningDetail}
          />
          <Route path={AppRoutes.VERGUNNINGEN} component={Vergunningen} />
          <Route component={NotFound} />
        </Switch>
      </div>
      <MainFooter />
    </>
  );
}

function AppLanding() {
  const session = useContext(SessionContext);

  const { isPristine, isAuthenticated, validityInSeconds } = session;

  // If session was previously authenticated we don't want to show the loader again
  if (isPristine) {
    return <p className={styles.PreLoader}>Welkom bij Mijn Amsterdam</p>;
  }

  const dialogTimeoutSettings = {
    secondsBeforeDialogShow:
      validityInSeconds ||
      DefaultAutologoutDialogSettings.secondsBeforeDialogShow,
  };

  // Render the main app only if we are authenticated
  return isAuthenticated ? (
    <>
      <AppAuthenticated />
      <AutoLogoutDialog settings={dialogTimeoutSettings} />
    </>
  ) : (
    <AppNotAuthenticated />
  );
}

export default function App() {
  useAnalytics(!!getOtapEnvItem('analyticsId'));
  useScript('/js/usabilla.js', false, true, IS_PRODUCTION);

  const sendToSentry = (error: Error, componentStack: string) => {
    Sentry.captureException(error, {
      extra: {
        componentStack,
      },
    });
  };

  return (
    <RecoilRoot>
      <BrowserRouter>
        <ErrorBoundary
          onError={sendToSentry}
          FallbackComponent={ApplicationError}
        >
          <SessionState>
            <AppLanding />
          </SessionState>
        </ErrorBoundary>
      </BrowserRouter>
    </RecoilRoot>
  );
}
