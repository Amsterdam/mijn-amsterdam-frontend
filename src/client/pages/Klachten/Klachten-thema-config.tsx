import { Klacht } from '../../../server/services/klachten/types';
import { AppRoutes } from '../../../universal/config/routes';
import { LinkProps } from '../../../universal/types/App.types';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

const MAX_TABLE_ROWS_ON_THEMA_PAGINA_LOPEND = 5;

export const LinkListItems: LinkProps[] = [
  {
    title: 'Meer informatie over de afhandeling van uw klacht',
    to: 'https://www.amsterdam.nl/veelgevraagd/klacht-indienen-over-de-gemeente-42fd5#case_%7B9846AD0A-E989-4B5D-A1D3-6D79E34DF1BE%7D',
  },
];

const displayPropsKlachten: DisplayProps<WithDetailLinkComponent<Klacht>> = {
  detailLinkComponent: 'Nummer van uw klacht',
  ontvangstDatumFormatted: 'Ontvangen op',
  onderwerp: 'Onderwerp',
};

export const routes = {
  listPage: AppRoutes['KLACHTEN/LIST'],
  detailPageVergunning: AppRoutes['KLACHTEN/KLACHT'],
  themaPage: AppRoutes.KLACHTEN,
} as const;

export const klachtenTableConfig = {
  title: 'Ingediende klachten',
  displayProps: displayPropsKlachten,
  maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_LOPEND,
  listPageRoute: routes.listPage,
} as const;
