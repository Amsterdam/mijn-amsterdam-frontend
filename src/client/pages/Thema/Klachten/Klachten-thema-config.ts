import { generatePath } from 'react-router';

import { KlachtFrontend } from '../../../../server/services/klachten/types';
import { LinkProps } from '../../../../universal/types/App.types';
import { DisplayProps } from '../../../components/Table/TableV2.types';
import { MAX_TABLE_ROWS_ON_THEMA_PAGINA_LOPEND } from '../../../config/app';
import type { ThemaRoutesConfig } from '../../../config/thema-types';

export const featureToggle = {
  klachtenActive: true,
};

export const themaId = 'KLACHTEN' as const;
export const themaTitle = 'Klachten';

export const routeConfig = {
  detailPage: {
    path: '/klachten/klacht/:id',
    trackingUrl: '/klachten/klacht',
    documentTitle: `Klachten | ${themaTitle}`,
  },
  listPage: {
    path: '/klachten/lijst/:page?',
    documentTitle: `Ingediende klachten | ${themaTitle}`,
  },
  themaPage: {
    path: '/klachten',
    documentTitle: `${themaTitle} | overzicht`,
  },
} as const satisfies ThemaRoutesConfig;

export const LinkListItems: LinkProps[] = [
  {
    title: 'Meer informatie over de afhandeling van uw klacht',
    to: 'https://www.amsterdam.nl/veelgevraagd/klacht-indienen-over-de-gemeente-42fd5#case_%7B9846AD0A-E989-4B5D-A1D3-6D79E34DF1BE%7D',
  },
];

const displayProps: DisplayProps<KlachtFrontend> = {
  props: {
    detailLinkComponent: 'Nummer van uw klacht',
    ontvangstDatumFormatted: 'Ontvangen op',
    onderwerp: 'Onderwerp',
  },
  colWidths: {
    large: ['25%', '25%', '50%'],
    small: ['50%', '50%', '0'],
  },
};

export const klachtenTableConfig = {
  title: 'Ingediende klachten',
  displayProps,
  maxItems: MAX_TABLE_ROWS_ON_THEMA_PAGINA_LOPEND,
  listPageRoute: generatePath(routeConfig.listPage.path, { page: null }),
} as const;
