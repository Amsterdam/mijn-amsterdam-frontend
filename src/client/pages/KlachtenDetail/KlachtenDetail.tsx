import { generatePath, useParams } from 'react-router-dom';

import type { Klacht } from '../../../server/services/klachten/types';
import { AppRoutes } from '../../../universal/config/routes';
import { isError, isLoading } from '../../../universal/helpers/api';
import { defaultDateFormat } from '../../../universal/helpers/date';
import {
  DetailPage,
  ErrorAlert,
  InfoDetail,
  PageContent,
  PageHeading,
  ThemaIcon,
} from '../../components';
import { ThemaTitles } from '../../config/thema';
import { useAppStateGetter } from '../../hooks/useAppState';
import { KLACHTEN_PAGE_SIZE } from '../Klachten/Klachten';

export default function KlachtenDetail() {
  const { KLACHTEN } = useAppStateGetter();
  const { id } = useParams<{ id: string }>();

  const klachtIndex = (KLACHTEN.content?.klachten || []).findIndex(
    (klacht: Klacht) => klacht.id === id
  );

  const klacht = KLACHTEN.content?.klachten[klachtIndex];

  const noContent = !isLoading(KLACHTEN) && !klacht;

  return (
    <DetailPage>
      <PageHeading
        icon={<ThemaIcon />}
        backLink={{
          to: generatePath(AppRoutes.KLACHTEN, {
            page:
              klachtIndex > 0
                ? Math.ceil((klachtIndex + 1) / KLACHTEN_PAGE_SIZE)
                : 1,
          }),
          title: ThemaTitles.KLACHTEN,
        }}
        isLoading={isLoading(KLACHTEN)}
      >
        {klacht?.onderwerp || 'Klacht'}
      </PageHeading>

      <PageContent>
        {isError(KLACHTEN) || noContent ? (
          <ErrorAlert>We kunnen op dit moment geen gegevens tonen.</ErrorAlert>
        ) : (
          <>
            <InfoDetail
              label="Nummer van uw klacht"
              value={klacht?.id || '-'}
            />
            <InfoDetail
              label="Ontvangen op"
              value={
                klacht?.ontvangstDatum
                  ? defaultDateFormat(klacht?.ontvangstDatum)
                  : '-'
              }
            />
            <InfoDetail
              label="Wat is de klacht?"
              value={klacht?.omschrijving}
            />
            {klacht?.locatie && (
              <InfoDetail
                label="Wat is de locatie waar de klacht is ontstaan?"
                value={klacht?.locatie}
              />
            )}
            {klacht?.gewensteOplossing && (
              <InfoDetail
                label="Wat wilt u dat de gemeente gaat doen naar aanleiding van uw klacht?"
                value={klacht?.gewensteOplossing}
              />
            )}
          </>
        )}
      </PageContent>
    </DetailPage>
  );
}
