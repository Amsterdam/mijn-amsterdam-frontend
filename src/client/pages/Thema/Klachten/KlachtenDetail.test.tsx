import { render } from '@testing-library/react';
import { generatePath } from 'react-router';
import { MutableSnapshot } from 'recoil';

import { routeConfig } from './Klachten-thema-config';
import { KlachtenDetail } from './KlachtenDetail';
import { AppState } from '../../../../universal/types/App.types';
import { appStateAtom } from '../../../hooks/useAppState';
import MockApp from '../../MockApp';

const testState = {
  KLACHTEN: {
    status: 'OK',
    content: [
      {
        gewensteOplossing: null,
        inbehandelingSinds: '2022-05-30T00:00:00.000Z',
        locatie: null,
        omschrijving: 'Dit is de omschrijving van de klacht',
        onderwerp: 'Test voor decentrale toewijzing',
        ontvangstDatum: '2022-05-30T00:00:00.000Z',
        id: '36049',
        link: {
          title: 'Klacht 36049',
          to: '/klachten/klacht/36049',
        },
      },
      {
        gewensteOplossing: null,
        inbehandelingSinds: '2022-05-18T00:00:00.000Z',
        locatie: null,
        omschrijving: 'Dear Amsterdam Municipality',
        onderwerp: null,
        ontvangstDatum: '2022-05-05T00:00:00.000Z',
        id: '36046',
        link: {
          title: 'Klacht 36046',
          to: '/klachten/klacht/36046',
        },
      },
      {
        inbehandelingSinds: '2022-04-14T00:00:00.000Z',
        ontvangstDatum: '2022-04-13T00:00:00.000Z',
        omschrijving: 'Een klacht.',
        gewensteOplossing: 'Boosterprik',
        onderwerp: 'Boosterprik',
        id: '34994',
        locatie: 'RAI',
        link: {
          to: '/klachten/klacht/34994',
          title: 'Klacht 34994',
        },
      },
    ],
  },
} as AppState;

function initializeState(snapshot: MutableSnapshot) {
  snapshot.set(appStateAtom, testState);
}

function setupMockComponent(id: string) {
  const routeEntry = generatePath(routeConfig.detailPage.path, {
    id,
  });
  const routePath = routeConfig.detailPage.path;

  return function Component() {
    return (
      <MockApp
        routeEntry={routeEntry}
        routePath={routePath}
        component={KlachtenDetail}
        initializeState={initializeState}
      />
    );
  };
}

describe('KlachtenDetail', () => {
  describe('No subject or location', () => {
    const Component = setupMockComponent('36046');

    it('Matches the Full Page snapshot', () => {
      const { asFragment } = render(<Component />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Subject no location', () => {
    const Component = setupMockComponent('36049');

    it('Matches the Full Page snapshot', () => {
      const { asFragment } = render(<Component />);
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('Subject and location', () => {
    const Component = setupMockComponent('34994');

    it('Matches the Full Page snapshot', () => {
      const { asFragment } = render(<Component />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
