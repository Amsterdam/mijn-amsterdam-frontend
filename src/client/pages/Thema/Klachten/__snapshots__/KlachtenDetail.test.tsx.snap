// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KlachtenDetail > No subject or location > Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <div
    class="ams-grid"
  >
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide _PageHeadingWrap_c5a3f1"
    >
      <div
        class="_PageHeadingInner_c5a3f1"
        id="skip-to-id-AppContent"
      >
        <nav
          class="ams-breadcrumb"
        >
          <ol
            class="ams-breadcrumb__list"
          >
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/klachten"
              >
                Klachten
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/klachten/klacht/36046"
                tabindex="-1"
              >
                <span
                  class="ams-visually-hidden"
                >
                  Klacht
                </span>
              </a>
            </li>
          </ol>
        </nav>
        <h1
          class="ams-heading ams-heading--1 _PageHeading_c5a3f1"
        >
          Klacht
        </h1>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <dl
        class="_Datalist_08bf39"
      >
        <dt
          class="_Datalist__title_08bf39"
        >
          Nummer van uw klacht
        </dt>
        <dd
          class="_Datalist__description_08bf39"
        >
          36046
        </dd>
        <dt
          class="_Datalist__title_08bf39"
        >
          Wat is de klacht?
        </dt>
        <dd
          class="_Datalist__description_08bf39"
        >
          Dear Amsterdam Municipality
        </dd>
      </dl>
    </div>
  </div>
</DocumentFragment>
`;

exports[`KlachtenDetail > Subject and location > Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <div
    class="ams-grid"
  >
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide _PageHeadingWrap_c5a3f1"
    >
      <div
        class="_PageHeadingInner_c5a3f1"
        id="skip-to-id-AppContent"
      >
        <nav
          class="ams-breadcrumb"
        >
          <ol
            class="ams-breadcrumb__list"
          >
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/klachten"
              >
                Klachten
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/klachten/klacht/34994"
                tabindex="-1"
              >
                <span
                  class="ams-visually-hidden"
                >
                  Boosterprik
                </span>
              </a>
            </li>
          </ol>
        </nav>
        <h1
          class="ams-heading ams-heading--1 _PageHeading_c5a3f1"
        >
          Boosterprik
        </h1>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <dl
        class="_Datalist_08bf39"
      >
        <dt
          class="_Datalist__title_08bf39"
        >
          Nummer van uw klacht
        </dt>
        <dd
          class="_Datalist__description_08bf39"
        >
          34994
        </dd>
        <dt
          class="_Datalist__title_08bf39"
        >
          Wat is de klacht?
        </dt>
        <dd
          class="_Datalist__description_08bf39"
        >
          Een klacht.
        </dd>
        <dt
          class="_Datalist__title_08bf39"
        >
          Wat is de locatie waar de klacht is ontstaan?
        </dt>
        <dd
          class="_Datalist__description_08bf39"
        >
          RAI
        </dd>
        <dt
          class="_Datalist__title_08bf39"
        >
          Wat wilt u dat de gemeente gaat doen naar aanleiding van uw klacht?
        </dt>
        <dd
          class="_Datalist__description_08bf39"
        >
          Boosterprik
        </dd>
      </dl>
    </div>
  </div>
</DocumentFragment>
`;

exports[`KlachtenDetail > Subject no location > Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <div
    class="ams-grid"
  >
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide _PageHeadingWrap_c5a3f1"
    >
      <div
        class="_PageHeadingInner_c5a3f1"
        id="skip-to-id-AppContent"
      >
        <nav
          class="ams-breadcrumb"
        >
          <ol
            class="ams-breadcrumb__list"
          >
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/klachten"
              >
                Klachten
              </a>
            </li>
            <li
              class="ams-breadcrumb__item"
            >
              <a
                class="ams-breadcrumb__link"
                href="/klachten/klacht/36049"
                tabindex="-1"
              >
                <span
                  class="ams-visually-hidden"
                >
                  Test voor decentrale toewijzing
                </span>
              </a>
            </li>
          </ol>
        </nav>
        <h1
          class="ams-heading ams-heading--1 _PageHeading_c5a3f1"
        >
          Test voor decentrale toewijzing
        </h1>
      </div>
    </div>
    <div
      class="ams-grid__cell ams-grid__cell--span-4 ams-grid__cell--span-8-medium ams-grid__cell--span-12-wide ams-grid__cell--start-1 ams-grid__cell--start-1-medium ams-grid__cell--start-1-wide"
    >
      <dl
        class="_Datalist_08bf39"
      >
        <dt
          class="_Datalist__title_08bf39"
        >
          Nummer van uw klacht
        </dt>
        <dd
          class="_Datalist__description_08bf39"
        >
          36049
        </dd>
        <dt
          class="_Datalist__title_08bf39"
        >
          Wat is de klacht?
        </dt>
        <dd
          class="_Datalist__description_08bf39"
        >
          Dit is de omschrijving van de klacht
        </dd>
      </dl>
    </div>
  </div>
</DocumentFragment>
`;
